notebooks/02_quant_activation_overview.ipynb

```diff
@@ -664,25 +664,233 @@
+       "    (fused_activation_quant_proxy): FusedActivationQuantProxy(\n",
+       "      (activation_impl): ReLU()\n",
+       "      (tensor_quant): RescalingIntQuant(\n",
+       "        (int_quant): IntQuant(\n",
+       "          (float_to_int_impl): RoundSte()\n",
+       "          (tensor_clamp_impl): TensorClamp()\n",
+       "          (delay_wrapper): DelayWrapper(\n",
+       "            (delay_impl): _NoDelay()\n",
+       "          )\n",
+       "        )\n",
+       "        (scaling_impl): ParameterFromRuntimeStatsScaling(\n",
+       "          (stats_input_view_shape_impl): OverTensorView()\n",
+       "          (stats): _Stats(\n",
+       "            (stats_impl): AbsPercentile()\n",
+       "          )\n",
+       "          (restrict_scaling): _RestrictValue(\n",
+       "            (restrict_value_impl): FloatRestrictValue()\n",
+       "          )\n",
+       "          (clamp_scaling): _ClampValue(\n",
+       "            (clamp_min_ste): ScalarClampMinSte()\n",
+       "          )\n",
+       "          (restrict_inplace_preprocess): Identity()\n",
+       "          (restrict_preprocess): Identity()\n",
+       "        )\n",
+       "        (int_scaling_impl): IntScaling()\n",
+       "        (zero_point_impl): ZeroZeroPoint(\n",
+       "          (zero_point): StatelessBuffer()\n",
+       "        )\n",
+       "        (msb_clamp_bit_width_impl): BitWidthConst(\n",
+       "          (bit_width): StatelessBuffer()\n",
+       "        )\n",
+       "      )\n",
+       "    )\n",
+       "  )\n",
+       ")"
+      ]
+     },
+     "execution_count": 33,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "per_tensor_quant_relu"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we initialise a new `QuantRelU` instance, but this time we specify that we desire per-channel quantization i.e. `scaling_per_output_channel=True`. This will implictly call `scaling_stats_input_view_shape_impl`, defined [here](https://github.com/Xilinx/brevitas/blob/200456825f3b4b8db414f2b25b64311f82d3991a/src/brevitas/quant/solver/common.py#L184), and will change the `QuantReLU` from using a per-tensor view when gathering stats to a per output channel view ([`OverOutputChannelView`](https://github.com/Xilinx/brevitas/blob/200456825f3b4b8db414f2b25b64311f82d3991a/src/brevitas/core/function_wrapper/shape.py#L52)). This simply permutes the tensor into a 2D tensor, with dim 0 equal to the number of output channels.\n",
+    "\n",
+    "To accomplish this, we also need to give it some extra information: `scaling_stats_permute_dims` and `per_channel_broadcastable_shape`. `scaling_stats_permute_dims` is responsible for defining how we do the permutation. `per_channel_broadcastable_shape` simply represents what the dimensions of the quantization parameters will be, i.e. there should be one parameter per output channel.\n",
```

Comment:
`per_channel_broadcastable_shape` is necessary to understand which along which dimensions the scale factor has to be broadcasted to match the dimensions of the input.
By default, pytorch will broadcast along the first rightmost dimension for which the shapes of the two tensors matches. To make sure that we apply the scale factor in our desired dimension, we need to tell PyTorch how to correctly broadcast the scale factors.

Reply:
Awesome, will add more context!

Reply:
I took your description, as it was a very good explanation haha, and added it to the PR!